qiskit/quantum_info/synthesis/clifford_decompose.py

```diff
@@ -12,25 +12,12 @@
 """
 Circuit synthesis for the Clifford class.
 """
-# pylint: disable=invalid-name
 
-from itertools import product
-
-import numpy as np
-
-from qiskit.circuit import QuantumCircuit
-from qiskit.exceptions import QiskitError
-from qiskit.quantum_info.operators.symplectic.clifford_circuits import (
-    _append_cx,
-    _append_h,
-    _append_s,
-    _append_swap,
-    _append_v,
-    _append_w,
-    _append_x,
-    _append_z,
+from qiskit.synthesis.clifford import (
+    synth_clifford_ag,
+    synth_clifford_bm,
+    synth_clifford_greedy,
 )
-from qiskit.quantum_info.operators.symplectic.pauli import Pauli
 
 
 def decompose_clifford(clifford, method=None):
```

Comment:
Why not transfer this function to `qiskit.synthesis`?

Reply:
As you suggested, I will transfer `decompose_clifford` into `qiskit.synthesis` and add a deprecation warning.